debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/outbox/MongoEventRouterTest.java

```diff
@@ -0,0 +1,979 @@
+        config.put(
+                MongoEventRouterConfigDefinition.ROUTE_TOMBSTONE_ON_EMPTY_PAYLOAD.name(),
+                "true");
+        router.configure(config);
+
+        final Map<String, Object> extraValues = new HashMap<>();
+        extraValues.put("is_deleted", true);
+
+        final SourceRecord eventRecord = createEventRecord(
+                "000000000000000000000000",
+                "UserCreated",
+                "10711fa5",
+                "User",
+                new Document(),
+                extraValues);
+        final SourceRecord eventRouted = router.apply(eventRecord);
+
+        Struct value = (Struct) eventRouted.value();
+        assertThat(value).isNotNull();
+        assertThat(value.get("deleted")).isEqualTo(true);
+
+        final SourceRecord eventRecordTombstone = createEventRecord(
+                "000000000000000000000000",
+                "UserCreated",
+                "10711fa5",
+                "User",
+                null,
+                extraValues);
+        final SourceRecord eventRoutedTombstone = router.apply(eventRecordTombstone);
+
+        Struct tombstone = (Struct) eventRoutedTombstone.value();
+        assertThat(tombstone).isNull();
+        VerifyRecord.isValidTombstone(eventRoutedTombstone);
+    }
+
+    @Test
+    public void noTombstoneIfNotConfigured() {
+        final Map<String, String> config = new HashMap<>();
+        config.put(
+                MongoEventRouterConfigDefinition.FIELDS_ADDITIONAL_PLACEMENT.name(),
+                "is_deleted:envelope:deleted");
+        router.configure(config);
+
+        final Map<String, Object> extraValues = new HashMap<>();
+        extraValues.put("is_deleted", true);
+
+        final SourceRecord eventRecord = createEventRecord(
+                "000000000000000000000000",
+                "UserCreated",
+                "10711fa5",
+                "User",
+                new Document(),
+                extraValues);
+        final SourceRecord eventRouted = router.apply(eventRecord);
+
+        Struct value = (Struct) eventRouted.value();
+        assertThat(value).isNotNull();
+        assertThat(value.get("deleted")).isEqualTo(true);
+
+        final SourceRecord eventRecordTombstone = createEventRecord(
+                "000000000000000000000001",
+                "UserCreated",
+                "10711fa5",
+                "User",
+                null,
+                extraValues);
+        final SourceRecord eventRoutedTombstone = router.apply(eventRecordTombstone);
+
+        Struct tombstone = (Struct) eventRoutedTombstone.value();
+        assertThat(eventRoutedTombstone.key()).isNotNull();
+        assertThat(eventRoutedTombstone.keySchema()).isNotNull();
+        assertThat(tombstone).isNotNull();
+        assertThat(tombstone.get("deleted")).isEqualTo(true);
+        assertThat(eventRoutedTombstone.valueSchema()).isNotNull();
+    }
+
+    @Test
+    public void canExpandJSONPayloadIfConfigured() {
```

Comment:
```suggestion
    public void canExpandJsonPayloadIfConfigured() {
```

(Here and I suppose elsewhere too). We use that style for abbreviations in (non-static) member names.